File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
"""
Time = O(n)
Space = O(1)
"""
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        # Initialize two pointers
        left = 0
        right = len(numbers) - 1
        
        # Traverse from the leftmost and the rightmost
        while left < right:
            cur_sum = numbers[left] + numbers[right]
            # Sum is equals to the target
            if cur_sum == target:
                return [left + 1, right + 1]
            # Sum is smaller than the target
            elif cur_sum < target:
                left += 1
            # Sum is larger than the target
            else:
                right -= 1
        
        return [-1, -1]

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int left = 0, right = numbers.size() - 1;
        int curSum = 0;
        vector<int> res = {-1, -1};

        while (left < right) {
            curSum = numbers[left] + numbers[right];
            if (curSum > target) {
                right--;
            } else if (curSum < target) {
                left++;
            } else {
                res[0] = left + 1;
                res[1] = right + 1;
                return res;
            }
        }

        return res;
    }
};